File: src/dataset/cpp/armstrong-numbers_exercism-tqa236/openAI/ChatGPT4_armstrong_numbers_v1.cpp
#include "armstrong_numbers.h"

bool armstrong_numbers::is_armstrong_number(int number) {
  int length = 0, number_copy = number, armstrong_sum = 0;

  // Calculate the number of digits
  for (int temp = number; temp > 0; temp /= 10) {
    ++length;
  }

  // Calculate Armstrong sum
  while (number_copy > 0) {
    int digit = number_copy % 10;
    number_copy /= 10;

    int power = 1;
    for (int i = 0; i < length; ++i) {
      power *= digit;
    }

    armstrong_sum += power;
    if (armstrong_sum > number) { // Early exit if sum exceeds the number
      return false;
    }
  }

  return armstrong_sum == number;
}

File: src/dataset/typescript/armstrong-numbers_exercism-samajammin/openAI/ChatGPT4_armstrong-numbers_v1.ts
export default class ArmstrongNumbers {
  static isArmstrongNumber(num: number): boolean {
    const digits = num.toString();
    const length = digits.length;
    let sum = 0;

    for (let i = 0; i < length; i++) {
      const digit = Number(digits[i]);
      sum += digit ** length;
      if (sum > num) return false; // Early exit if sum exceeds num
    }

    return sum === num;
  }
}